front/app/modules/commercial/user_custom_fields/admin/containers/CustomFields/RegistrationCustomFieldForm/index.tsx

```diff
@@ -31,131 +29,105 @@ export interface FormValues {
   required: boolean;
 }
 
-export interface Props {
+type Props = {
   mode: 'new' | 'edit';
-  customFieldId: string;
+  customFieldId?: string;
   builtInField: boolean;
-}
+  defaultValues?: FormValues;
+  onSubmit: (formValues: FormValues) => void | Promise<void>;
+} & InjectedIntlProps;
 
-class RegistrationCustomFieldForm extends React.Component<
-  InjectedFormikProps<Props & InjectedIntlProps, FormValues>
-> {
-  public static validate = (values: FormValues): FormikErrors<FormValues> => {
-    const errors: FormikErrors<FormValues> = {};
+const fieldTypes = [
```

Comment:
I noticed that these values are similar to [these](https://github.com/CitizenLabDotCo/citizenlab/blob/master/front/app/modules/commercial/user_custom_fields/services/userCustomFields.ts#L9-L16)

Maybe we could keep it dry by doing something like.
```
const fieldTypes = [
  'select',
  'multiselect',
  'checkbox',
  'text',
  'multiline_text',
  'number',
  'date',
] as const;
export type IUserCustomFieldInputType = typeof fieldTypes[number];
```